antarest/common/config.py

```diff
@@ -1,33 +1,50 @@
 import os
 from copy import deepcopy
+from dataclasses import dataclass
 from pathlib import Path
-from typing import Any, Optional
+from typing import Any, Optional, Dict, Type, TypeVar, Callable, cast
 
 import yaml
+from dataclasses_json import DataClassJsonMixin  # type: ignore
+from dataclasses_json.api import dataclass_json
 
 from antarest.common.custom_types import JSON
 
+config_definitions: Dict[str, Type[DataClassJsonMixin]] = {}
+
 
 class Config:
     def __init__(self, data: Optional[JSON] = None):
-        self.data = data or dict()
+        self.raw_data = data or dict()
+        self.data: Dict[str, Any] = {}
+        self._init()
 
     def __getitem__(self, item: str) -> Any:
         return self._get(item)
 
+    def _init(self) -> None:
+        global config_definitions
```

Comment:
can we do without global ?

Reply:
as it is a idiom in python to have singleton as module, this is why this is a global (module scoped) variable